src/Microsoft.DotNet.Cli.Utils/CommandResolution/MSBuildProject.cs

```diff
@@ -135,7 +135,7 @@ private string GetLockFilePathFromProjectLockFileProperty()
                 .AllEvaluatedProperties
                 .Where(p => p.Name.Equals("ProjectAssetsFile"))
                 .Select(p => p.EvaluatedValue)
-                .FirstOrDefault(p => File.Exists(p));
+                .FirstOrDefault(p => Path.IsPathRooted(p) && File.Exists(p));
```

Comment:
Because without a full path we fail to open the lock file (I don't know how NuGet is resolving relative). Before my change we only had a full path. But because the test project is csproj the query now returns a relative path too. But we need to the full path to have NuGet open it. @livarcocc might have a more complete explanation :)